src/utils/typedefs-wrapper.cpp

```diff
@@ -44,26 +46,286 @@ namespace detail {
     }
     Il2CppString* alloc_str(std::u16string_view str) {
         il2cpp_functions::Init();
-        return il2cpp_functions::string_new_utf16((Il2CppChar*) str.data(), str.size());
+        return il2cpp_functions::string_new_utf16((Il2CppChar const*) str.data(), str.size());
     }
 }
 }
 
-StringW::operator std::string() {
+StringW::operator std::string() const {
     std::string val(inst->length * 2, '\0');
     auto resSize = il2cpp_utils::detail::convstr(inst->chars, val.data(), inst->length, val.size());
     val.resize(resSize);
     return val;
 }
 
-StringW::operator std::u16string() {
+StringW::operator std::u16string() const {
     return {inst->chars, inst->chars + inst->length};
 }
 
-StringW::operator std::wstring() {
+StringW::operator std::wstring() const {
     return {inst->chars, inst->chars + inst->length};
 }
 
+StringW::operator const std::u16string_view() const {
+    return {inst->chars, static_cast<std::size_t>(inst->length)};
+}
+
 StringW::operator std::u16string_view() {
     return {inst->chars, static_cast<std::size_t>(inst->length)};
 }
+
+bool strcomp(Il2CppString const* lhs, const std::u16string_view rhs) noexcept {
+    if (!lhs || lhs->length != (int)rhs.size()) return false;
```

Comment:
Is this `int` cast necessary? I would hope not, but I don't remember max length of a string, perhaps it makes more sense to cast to `decltype(lhs->length)`? not sure though